src/graph/visitor/PropertyTrackerVisitor.cpp

```diff
@@ -208,10 +208,19 @@ void PropertyTrackerVisitor::visit(AttributeExpression *expr) {
   auto &propName = constVal.getStr();
   switch (lhs->kind()) {
     case Expression::Kind::kInputProperty:
-    case Expression::Kind::kVarProperty: {  // $e.name
+    case Expression::Kind::kVarProperty: {
+      // maybe: $e.prop
       auto *varPropExpr = static_cast<PropertyExpression *>(lhs);
-      auto &edgeAlias = varPropExpr->prop();
-      propsUsed_.insertEdgeProp(edgeAlias, unknownType_, propName);
+      auto &entityAlias = varPropExpr->prop();
+      propsUsed_.insertEdgeProp(entityAlias, unknownType_, propName);
+      // maybe: $v.tag
+      auto ret = qctx_->schemaMng()->toTagID(space_, propName);
+      if (!ret.ok()) {
+        status_ = std::move(ret).status();
+        break;
+      }
+      auto tagId = ret.value();
+      propsUsed_.insertVertexProp(entityAlias, tagId, "*");
```

Comment:
Is it necessary to insert edge and vertex props at same time? maybe only need to handle one branch by checking where the prop is from?

Reply:
It goes into `Kind::kSubscript` as `$-.e[0]`. Maybe we can delete it already.

Reply:
Seems like we don't have sufficient information here to tell them apart. Any advise?